Tagging UX refinement v2 [FC-0036]

ChrisChV opened the pull request:
<!--

🌳🌳
🌳🌳🌳🌳         🌳 Note: Quince is in support. Fixes you make on master may still be needed on Quince.
    🌳🌳🌳🌳     If so, make another pull request against the open-release/quince.master branch,
🌳🌳🌳🌳         or ask in the #wg-build-test-release Slack channel if you have any questions or need help.
🌳🌳


Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly
readable.  If you must linked information must be private (because it has secrets),
clearly label the link as private.

-->

## Description

This PR fixes the following issues:

1. the layer over the course outline is too dark when the tag drawer is open.
2. Refresh the count of tags on the unit/outline page(s) after the user makes edits in the tag drawer
    - TagList component updated to allow receiving for the Manage tags drawer and refresh all the tag tree and counts
    - TagCount refactored to use javascript to render the component and allow receiving for the Manage tags drawer and refresh the count.
    - Update new TagCount in units and components

## Supporting information

Internal ticket: [FAL-3601](https://tasks.opencraft.com/browse/FAL-3601)
Github issue: https://github.com/openedx/modular-learning/issues/167
Related PR: https://github.com/openedx/frontend-app-course-authoring/pull/800

## Testing instructions

Enable `contentstore.enable_copy_paste_units` and `new_studio_mfe.use_tagging_taxonomy_list_page flags`.
Before to all you need to run the [taxonomy-sample-data](https://github.com/open-craft/taxonomy-sample-data).

**1. the layer over the course outline is too dark when the tag drawer is open.**

- Go to a [sample course](http://localhost:18010/course/course-v1:SampleTaxonomyOrg2+STC1+2023_1) and go to a unit. Click on Manage tags to open the drawer. Verify the color.

**2. Refresh the count of tags on the unit/outline page(s) after the user makes edits in the tag drawer**

- Use this branch of [frontend-app-course-authoring](https://github.com/openedx/frontend-app-course-authoring/pull/800)-
- Go to a [sample course](http://localhost:18010/course/course-v1:SampleTaxonomyOrg2+STC1+2023_1) and go to a unit. Click on Manage tags to open the drawer.
- Add and remove tags.
- Close the drawer and verify that the TagList is updated.
- Open the manage tags drawer for a component of the unit.
- Add and remove tags.
- Verify that the count of tags is updated.
- Go to a  [sample course](http://localhost:18010/course/course-v1:SampleTaxonomyOrg2+STC1+2023_1) and open the outline.
- Click on the tag count of any unit and open the manage tags drawer.
- Add and remove tags
- Verify that the count of tags is updated.

ChrisChV committed:
style: drawer-cover color updated for tagging drawer

ChrisChV committed:
style: drawer-cover color updated for all tagging drawers

openedx-webhooks commented:
Thanks for the pull request, @ChrisChV! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

ChrisChV committed:
feat: Update TagList component when a tag is updated on Manage tags drawer

ChrisChV committed:
feat: Reafactor TagCount to be hable to refresh the count

ChrisChV committed:
feat: Sync tag count in units

yusuf-musleh reviewed (approved):
:+1: The updated implementation works quite well, just had a suggestion about the messaging format in the related (https://github.com/openedx/frontend-app-course-authoring/pull/800) for the close message, which would need a small change here as well.

Great work @ChrisChV ! It's great seeing how all of this is coming together.

- [X] I tested this: I confirmed that the dark background color has changed to the lighter one. I also tested that when you open the manage tags drawer for a unit from the course outline, add/remove tags updates automatically on the course outline page, including hiding/showing the tags count when 0 or more that 0 tags applied. I also tested the same for opening the tags drawer for a unit from inside the unit page, and the updated tags are reflected automatically in the TagList on the side of the page. And finally I tested that accessing the manage tags drawer from a component in a unit also reflects the changes on the page when adding/removing tags (including hiding and showing the tags count)
- [X] I read through the code
- [X] I checked for accessibility issues
- [X] Includes documentation
- [ ] ~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~

bradenmacdonald commented:
@ChrisChV There is some issue with the JS in this PR. When I have it checked out, I can't build the tutor image - it fails with some error in the `r.js optimizer`. Perhaps you have included some new JS syntax that breaks the (super old) bundling code we have.

Related, I'm seeing this error on the Unit page, even if I run the rebuilt assets command.

![Screenshot 2024-01-22 at 12 27 29 PM](https://github.com/openedx/edx-platform/assets/945577/ae32db30-ab1d-4af3-beff-2b2f1d4a53db)

bradenmacdonald commented on cms/static/js/factories/tag_count.js:
```diff
@@ -0,0 +1,13 @@
+    tagCountView.render();
+}
+
+export {TagCountFactory};
```
Ah, these `import` and `export` statements are probably the problem - I don't think the build process used in edx-platform supports them. But it could be something else - not sure.

ChrisChV replied:
There are other factories that has the [same syntax](https://github.com/openedx/edx-platform/blob/61bb17650533192ff3bca95e848142234ffb90b9/cms/static/js/factories/tag_count.js). My tutor is unconfigured, I'm going to try to configure it to try another syntax

ChrisChV replied:
That error happened to me, but it was fixed with [this line](https://github.com/openedx/edx-platform/pull/34059/files#diff-fbed72c8f21e30539a12e9e1dfdff135c75340e4ca10894890d1a21c6a0988d5R87). Maybe clearing the cache?

ChrisChV replied:
Nevermind, I found the `r.js optimizer` error

bradenmacdonald replied:
Thx, let me know when you've found a fix.

ChrisChV replied:
@bradenmacdonald I fixed the bug. It was an extrra `,` [here](https://github.com/openedx/edx-platform/pull/34059/commits/6b78dc94247f89a7d850b3d80088d6a42a70938b#diff-ba366278f32578c3c9ad5b048f4d212a668e20c4abaf89eed318593b1006a600L36) You can test it? I still need to configure my tutor with the MFE

ChrisChV committed:
test: Fixing python tests

ChrisChV committed:
chore: Test to fix javascript error

chore: Test to fix javascript error

chore: Test to fix javascript error

chore: Test to fix javascript error

ChrisChV committed:
fix: Bug with javascript

bradenmacdonald commented:
Hmm, I can see the lighter color, and now the JS build error is resolved, but the auto-refresh isn't working for me. I see that Course-Authoring sends the signals, but the CMS is not refreshing anything.

@yusuf-musleh could you please confirm it's working for you in the latest version?

If so, I'm good to merge this as-is. We're actually going to shift focus to the new MFE versions of these pages very soon, so I don't want to spend too much more time on the "legacy" views.

yusuf-musleh commented:
@bradenmacdonald @ChrisChV I tried it out with the latest changes on this PR and the mfe PR, and ran through all the cases and can confirm that they're all working as expected on my devstack.

openedx-webhooks commented:
@ChrisChV 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/34059&entry.1484655318=https://github.com/ChrisChV&entry.752974735=2024-01-15+21:03&entry.1917517419=2024-01-25+18:33&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
